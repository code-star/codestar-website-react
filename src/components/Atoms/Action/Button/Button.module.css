.button {
	min-width: 160px;
	min-height: 30px;
	border-radius: 5px;
}
.buttonPrimary {
	color: #fff;
	background-color: #00f;
}
.buttonSecondary {
	color: #fff;
	background-color: #f00;
}
.buttonTertiary {
	color: #000;
	background-color: #ff0;
}
