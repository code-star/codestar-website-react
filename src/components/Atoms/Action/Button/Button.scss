@import '../../../../theming/config.scss';

.button {
	min-width: 160px;
	min-height: 30px;

	border-radius: 5px;

	&Primary {
		color: $color-white;
		background-color: $color-blue;
	}

	&Secondary {
		color: $color-white;
		background-color: $color-red;
	}

	&Tertiary {
		color: $color-black;
		background-color: $color-yellow;
	}
}
