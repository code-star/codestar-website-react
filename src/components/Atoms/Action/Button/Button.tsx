import React from 'react';
import classNames from 'classnames/bind';
import style from './Button.module.css';

const cx = classNames.bind(style);

interface IButtonProps {
  variant: string;
  children: any;
}

const Button = ({ variant, children }: IButtonProps) => {
  return (
    <button
      className={cx('button', {
        buttonPrimary: variant === 'primary',
        buttonSecondary: variant === 'secondary',
        buttonTertiary: variant === 'tertiary',
      })}
    >
      {children}
    </button>
  );
};

export default Button;
