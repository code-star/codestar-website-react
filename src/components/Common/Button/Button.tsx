import React, { SFC } from 'react';
import classNames from 'classnames/bind';
import style from './Button.module.css';

const cx = classNames.bind(style);

interface IButtonProps {
	variant: string;
	children: any;
}

const Button: SFC<IButtonProps> = ({ variant, children }) => {
	return (
		<button
			className={cx('button', {
				buttonPrimary: variant === 'primary',
				buttonSecondary: variant === 'secondary',
				buttonTertiary: variant === 'tertiary',
			})}
		>
			{children}
		</button>
	);
};

export default Button;
