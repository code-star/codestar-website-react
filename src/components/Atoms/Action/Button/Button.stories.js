import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs';
import Button from './Button.tsx';

storiesOf('Components/Atoms/Action', module)
	.addDecorator(withKnobs)
	.add('Button', () => {
		const variant = select(
			'Variant',
			['primary', 'secondary', 'tertiary'],
			'primary'
		);

		return (
			<Button variant={variant}>
				<span>Awesome {`${variant}`} button</span>
			</Button>
		);
	});
