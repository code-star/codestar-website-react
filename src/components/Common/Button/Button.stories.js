import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select, text } from '@storybook/addon-knobs';
import Button from './Button.tsx';

storiesOf('Components/Action', module)
	.addDecorator(withKnobs)
	.add('Button', () => {
		const variant = select(
			'Variant',
			['primary', 'secondary', 'tertiary'],
			'primary'
		);

		return (
			<Button variant={variant}>
				<span>Awesome {`${variant}`} button</span>
			</Button>
		);
	});
