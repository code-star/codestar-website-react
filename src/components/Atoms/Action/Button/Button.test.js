import React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';

const children = <span>awesome button</span>;

const getButton = variant => <Button variant={variant}>{children}</Button>;

const renderShallow = () => {
	return shallow(getButton('primary'));
};

describe.only('<Button />', () => {
	let wrapper;

	describe('Instance', () => {
		test('must be an instance of Button', () => {
			wrapper = renderShallow();
			expect(wrapper.find('Button')).toBeTruthy();
		});

		test('must render children', () => {
			wrapper = renderShallow();
			expect(wrapper).toContainReact(children);
		});
	});

	describe('Snaphot', () => {
		test('must match primary', () => {
			expect(global.renderToJSON(getButton('primary'))).toMatchSnapshot();
		});

		test('must match secondary', () => {
			expect(global.renderToJSON(getButton('secondary'))).toMatchSnapshot();
		});

		test('must match tertiary', () => {
			expect(global.renderToJSON(getButton('tertiary'))).toMatchSnapshot();
		});
	});
});
