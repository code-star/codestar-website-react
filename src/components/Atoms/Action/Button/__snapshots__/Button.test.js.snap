// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snaphot must match primary 1`] = `
<button
  className="button buttonPrimary"
>
  <span>
    Awesome button
  </span>
</button>
`;

exports[`<Button /> Snaphot must match secondary 1`] = `
<button
  className="button buttonSecondary"
>
  <span>
    Awesome button
  </span>
</button>
`;

exports[`<Button /> Snaphot must match tertiary 1`] = `
<button
  className="button buttonTertiary"
>
  <span>
    Awesome button
  </span>
</button>
`;
